apps/api/src/app/notifications/usecases/get-activity-feed/get-activity-feed.usecase.ts

```diff
@@ -46,6 +65,72 @@ export class GetActivityFeed {
     };
   }
 
+  private async validateRetentionLimitForTier(organizationId: string, after?: string, before?: string) {
+    const organization = await this.organizationRepository.findById(organizationId);
+
+    if (!organization) {
+      throw new HttpException('Organization not found', HttpStatus.INTERNAL_SERVER_ERROR);
+    }
+
+    const maxRetentionMs = this.getMaxRetentionPeriodByOrganization(organization);
+
+    const earliestAllowedDate = new Date(Date.now() - maxRetentionMs);
+
+    // If no after date is provided, default to the earliest allowed date
+    const effectiveAfterDate = after ? new Date(after) : earliestAllowedDate;
+    const effectiveBeforeDate = before ? new Date(before) : new Date();
+
+    this.validateDateRange(earliestAllowedDate, effectiveAfterDate, effectiveBeforeDate);
+
+    return {
+      after: effectiveAfterDate.toISOString(),
+      before: effectiveBeforeDate.toISOString(),
+    };
+  }
+
+  private validateDateRange(earliestAllowedDate: Date, afterDate: Date, beforeDate: Date) {
+    if (afterDate > beforeDate) {
+      throw new HttpException(
```

Comment:
Nice! @ChmaraX shouldn't we do these checks in GetActivityFeedCommand?

Reply:
I thought there is be a build-in validator from `class-validator` but there isn't and writing custom function seems to be slightly worse than having this simple check I guess.